File: QuickSort/Java/QuickSort.java
package QuickSort.Java;

import java.util.ArrayList;
import java.util.Arrays;

public class QuickSort {
    
    public static void main(String[] args) {
        ArrayList<Integer> arr = new ArrayList<Integer>(Arrays.asList(11,232,33,41,5,436,7,81,96,101));
        ArrayList<Integer> newArr = quickSort(arr);
        System.out.println(newArr);
    }

    public static ArrayList<Integer> quickSort(ArrayList<Integer> arr){
        
        int pivot;
        ArrayList<Integer> less = new ArrayList<Integer>();
        ArrayList<Integer> greater = new ArrayList<Integer>();
        if(arr.size()<2)
            return arr;
        else{
            pivot = arr.get(0);
            for(int i=1; i<arr.size();i++)
            {
                if(arr.get(i)<=pivot)
                {
                    less.add(arr.get(i));
                }                    
                else
                    greater.add(arr.get(i));   
            }
            ArrayList<Integer> newArr = quickSort(less);
            newArr.add(pivot);
            newArr.addAll(quickSort(greater));
            return newArr;  
        }
    }


}

File: QuickSort/C#/QuickSort.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace C_
{
    internal class QuickSort
    {
        public static void Main(string[] args)
        {
            ArrayList arr = new ArrayList() { 1, 23, 35, 12, 125, 1234, 54, 12, 431, 9 };
            ArrayList newArr = quickSort(arr);
            foreach (var i in newArr)
            {
                Console.WriteLine(i);
            }
        }

        public static ArrayList quickSort(ArrayList arr)
        {            
            dynamic pivot;
            ArrayList less = new ArrayList();
            ArrayList greater = new ArrayList();

            if(arr.Count < 2)
            {
                return arr;
            }
            else
            {
                pivot = arr[0];
                for (int i = 1; i < arr.Count; i++)
                {
                    if ((int)arr[i] <= pivot)                    
                        less.Add(arr[i]);                    
                    else
                        greater.Add(arr[i]);                  
                }

                ArrayList newArr = quickSort(less);
                newArr.Add(pivot);
                newArr.AddRange(quickSort(greater));
                return newArr;
            }
        }

    }
}
